Sources/SwiftlySearch/SwiftlySearch.swift

```diff
@@ -53,16 +69,23 @@ fileprivate struct SearchBar: UIViewControllerRepresentable {
         
         private var subscription: AnyCancellable?
         
-        init(text: Binding<String>) {
+        init(text: Binding<String>, placeholder: String?, hidesNavigationBarDuringPresentation: Bool, resultContent: (String) -> ResultContent?) {
             self._text = text
-            self.searchController = UISearchController(searchResultsController: nil)
+            
+            let resultView = resultContent(text.wrappedValue)
+            let searchResultController = resultView.map { UIHostingController(rootView: $0) }
+            self.searchController = UISearchController(searchResultsController: searchResultController)
             
             super.init()
             
             searchController.searchResultsUpdater = self
-            searchController.hidesNavigationBarDuringPresentation = true
+            searchController.hidesNavigationBarDuringPresentation = hidesNavigationBarDuringPresentation
             searchController.obscuresBackgroundDuringPresentation = false
             
+            if let placeholder = placeholder {
+                searchController.searchBar.placeholder = placeholder
+            }
```

Comment:
Since both `placeholder` and `searchController.searchBar.placeholder` are of type `String?`, I think this `if let` is redundant.

```suggestion
            searchController.searchBar.placeholder = placeholder
```

Reply:
I thought that too, but search bars actually don't have a default value of nil. Their default value is the localized search title. Setting it to nil will display an empty string.

Reply:
Ah, yeah that makes sense. Fair enough.